lib/buffer.js

```diff
@@ -370,8 +370,12 @@ Buffer.prototype.toString = function() {
   } else {
     var result = slowToString.apply(this, arguments);
   }
-  if (result === undefined)
-    throw new Error('toString failed');
+  if (result === undefined) {
+    var msg = 'toString failed - buffer larger than maximum string size ';
```

Comment:
Nit: `toString() failed: buffer larger than...` to match styling elsewhere in `lib` for error messages. (That's using `:` instead of `-` and putting `()` after `toString` to indicate it's a function.)